src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -111,7 +111,7 @@ public static function register($mode = 0)
             $trace = debug_backtrace(true);
             $group = 'other';
             $isVendor = false;
-            $isWeak = DeprecationErrorHandler::MODE_WEAK === $mode || (DeprecationErrorHandler::MODE_WEAK_VENDORS === $mode && $isVendor = $inVendors($file));
+            $isVendor = $inVendors($file);
```

Comment:
`$isVendor = DeprecationErrorHandler::MODE_WEAK_VENDORS === $mode && ...` (same below, so that we don't compute $isVendor when not needed, isn't it?)

Reply:
Yeah plus I have 2 `$isVendor` assignments now 😅

Reply:
The condition on line 147 can go